blocks/bricks/base.py

```diff
@@ -327,6 +328,16 @@ def copy_and_tag(variable, role, name):
         return unpack(outputs)
 
 
+# Ensure that Application instance are not pickled, and that when
+# unpickling, a reference to an existing Application object is
+# substituted instead of creating a new one.
```

Comment:
@dmitriy-serdyuk : if you have an idea how to do it with `__getstate__` and
`__setstate__`, I would be surely interested to listen to your proposal.
But as far as I understand, __setstate__ is only run *after* the object is
created, and there is no way to avoid creating a new object using
`__setstate__`.

On 3 February 2016 at 00:41, David Warde-Farley <notifications@github.com>
wrote:

> In blocks/bricks/base.py
> <https://github.com/mila-udem/blocks/pull/974#discussion_r51680836>:
>
> > @@ -327,6 +328,16 @@ def copy_and_tag(variable, role, name):
> >          return unpack(outputs)
> >
> >
> > +# Ensure that Application instance are not pickled, and that when
> > +# unpickling, a reference to an existing Application object is
> > +# substituted instead of creating a new one.
>
> Why doesn't it happen with regular pickling?
>
> Because the Application object is created at interpret-time, but when
> pickled it's serialized like everything else, so when you unpickle there's
> two.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mila-udem/blocks/pull/974/files#r51680836>.
>